roles/openshift_sanitize_inventory/tasks/deprecations.yml

```diff
@@ -1,21 +1,23 @@
 ---
 
 - name: Check for usage of deprecated variables
-  set_fact:
-    __deprecation_message: "{{ __deprecation_message | default( __deprecation_header ) }} \n\t{{ item }}"
-  when:
-  - hostvars[inventory_hostname][item] is defined
-  with_items: "{{ __warn_deprecated_vars }}"
+  check_deprecated:
+    facts: "{{ hostvars[inventory_hostname] }}"
+    vars: "{{ __warn_deprecated_vars }}"
+    header: "{{ __deprecation_header }}"
+  register: dep_check
+  run_once: true
 
 - block:
-  - debug: msg="{{__deprecation_message}}"
+  - run_once: true
```

Comment:
Will only show first host's message.

Reply:
There should only be one message... or is it possible that there are multiple inventory files that can be used for different hosts? we have `run_once: true` on the action_plugin so this would be the same for every host

Reply:
different groups can have different variables defined.  EG:
```ini
[masters:vars]
something_here: 1

[nodes:vars]
something_deprecated_here: 1
something_else: 1
```

Reply:
Additionally, you can define variables on a per-host basis.

If we use run_once: true on the action plugin, then we're only going to check the first host's variables.